tests/test_signal_handlers.py

```diff
@@ -0,0 +1,344 @@
+    mock_source_instance = mocker.Mock(name="test-source")
+    mock_job.translation_source.get_source_instance.return_value = mock_source_instance
+
+    create_landed_translation_task(
+        sender=Job,
+        instance=mock_job,
+        translation=mock_translation,
+    )
+
+    mock_create_from_source_and_translation.assert_called_once_with(
+        source_object=mock_source_instance,
+        translated_locale=mock_translation.target_locale,
+    )
+
+
+def test_create_landed_translation_task__disabled_by_settings(
+    mocker,
+    smartling_settings,
+):
+    smartling_settings.ADD_APPROVAL_TASK_TO_DASHBOARD = False
+
+    mock_logger_debug = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers.logger.debug"
+    )
+    mock_create_from_source_and_translation = mocker.patch(
+        "wagtail_localize_smartling.models.LandedTranslationTask.objects.create_from_source_and_translation"
+    )
+
+    mock_job = mocker.MagicMock(spec=Job)
+    mock_translation = mocker.MagicMock(spec=Translation)
+
+    create_landed_translation_task(
+        sender=Job, instance=mock_job, translation=mock_translation
+    )
+
+    mock_logger_debug.assert_called_once_with(
+        "Creation of a landed-translation task is disabled by settings"
+    )
+    assert not mock_create_from_source_and_translation.called
+
+
+def test__close_translation_landed_task__happy_path(mocker):
+    mock_task = mocker.MagicMock(spec=LandedTranslationTask)
+    mock_task.complete = mocker.MagicMock()
+
+    mock_tasks_filter = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers.LandedTranslationTask.objects.filter",
+        return_value=[mock_task],
+    )
+    mock_instance = mocker.Mock()
+    mock_instance.pk = 1234
+    mock_content_type = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers.ContentType.objects.get_for_model",
+        return_value=mocker.Mock(),
+    )
+
+    _close_translation_landed_task(mock_instance)
+
+    mock_content_type.assert_called_once_with(mock_instance)
+    mock_tasks_filter.assert_called_once_with(
+        content_type=mock_content_type.return_value,
+        object_id=1234,
+    )
+    mock_task.complete.assert_called_once()
+
+
+def test__close_translation_landed_task__no_tasks_found(mocker):
+    mock_tasks_filter = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers.LandedTranslationTask.objects.filter",
+        return_value=[],
+    )
+    mock_instance = mocker.Mock()
+    mock_instance.pk = 5432
+    mock_content_type = mocker.patch(
+        "wagtail_localize_smartling.signal_handlers.ContentType.objects.get_for_model",
+        return_value=mocker.Mock(),
+    )
+
+    _close_translation_landed_task(mock_instance)
+
+    mock_content_type.assert_called_once_with(mock_instance)
+    mock_tasks_filter.assert_called_once_with(
+        content_type=mock_content_type.return_value,
+        object_id=5432,
+    )
```

Comment:
Could throw an assert that `complete` is not called?

Reply:
Maybe I'm misunderstanding what you're saying, but there won't be a `Task` on which to call complete, no?

Reply:
You're right. I was just comparing the test above and looking at the similarities.